pandas/core/groupby/ops.py

```diff
@@ -738,7 +738,7 @@ def agg_series(self, obj: Series, func: F):
             # TODO: can we get a performant workaround for EAs backed by ndarray?
             return self._aggregate_series_pure_python(obj, func)
 
-        elif obj.index._has_complex_internals:
+        elif obj.index._has_complex_internals or obj.index.dtype == "object":
```

Comment:
A comment here might be helpful explaining why we should expect an object type index to raise in the fastpath

Reply:
Yes, definitely! Done